preprocessors/ner/ner.py

```diff
@@ -0,0 +1,219 @@
+        file.write(eval(str(binary)))
+
+
+"""
+Given a json from the raw data, this function will extract the alt text
+:my_json: json from raw data
+"""
+def get_alt(my_json):
+    context = my_json['context']
+    soup = BeautifulSoup(context, 'html.parser')
+    alt = soup.find('img', alt=True)['alt']
+    return alt
+
+
+"""
+Object to pass parameters to clipscore module
+"""
+class Namespace:
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+
+"""
+Function to extarct the NERs from a given english sentence, using the Stanford ner model
+:sentence: the sentence to check
+"""
+def stanford_ner(sentence, only_ner = True):
+    # Prepare NER tagger with english model
+    ner_tagger = StanfordNERTagger(model, jar, encoding='utf8')
+
+    # Tokenize: Split sentence into words
+    words = nltk.word_tokenize(sentence)
+
+    # Run NER tagger on words
+    words =  ner_tagger.tag(words)
+    
+    if only_ner:
+        rtn  = []
+        for i in words:
+            if len(i[1]) > 4:
+                rtn.append(i)
+        return rtn
+    
+    return words
+
+
+"""
+Function to find the index of a given substring (without whitespace) in a string, interms of words starting with index 1.
+e.g. find_index("Hello I'm Namdar.", "Namdar") = 3
+:text: the string to check
+:word: the word to check for
+"""
+def find_index(text, word):
+    index = 1
+    for i in text.split():
+        if word in i:
+            return index
+        index += 1
+    return -1
+
+
+@app.route('/preprocessor', methods=['POST', 'GET'])
+def main():
+    
+    logging.debug("Received request")
+
+    with open('./schemas/preprocessors/ner.schema.json') as jsonfile:
+        data_schema = json.load(jsonfile)
+
+    with open('./schemas/preprocessor-response.schema.json') as jsonfile:
+        schema = json.load(jsonfile)
+
+    with open('./schemas/definitions.json') as jsonfile:
+        definition_schema = json.load(jsonfile)
+
+    with open('./schemas/request.schema.json') as jsonfile:
+        first_schema = json.load(jsonfile)
+
+    schema_store = {
+        data_schema['$id']: data_schema,
+        schema['$id']: schema,
+        definition_schema['$id']: definition_schema
+    }
+
+    resolver = jsonschema.RefResolver.from_schema(
+        schema, store=schema_store)
+
+    content = request.get_json()
+
+    try:
+        validator = jsonschema.Draft7Validator(first_schema, resolver=resolver)
+        validator.validate(content)
+    except jsonschema.exceptions.ValidationError as e:
+        logging.error(e)
+        return jsonify("Invalid Preprocessor JSON format"), 400
+
+    # ------ START COMPUTATION ------ #
+
+    name_ = "1"
+    captions = {}                            # dict to store the captions
+    text = get_alt(content)                  # extract alt text and save it in the caption dict
+    captions[name_] = text
+    html_ = content['graphic'] # get the image html content
```

Comment:
Note: you are *not* guaranteed to get a `graphic` key! Many IMAGE requests do not have a blob (embedded maps and embedded charts) and will throw an error here. Please check if the graphic tag exists, return a 204 code if it does not, and then continue on as normal.